railties/lib/rails/test_unit/test_parser.rb

```diff
@@ -13,29 +13,32 @@ module TestUnit
       # Parse a test file to extract the line ranges of all tests in both
       # method-style (def test_foo) and declarative-style (test "foo" do)
       module TestParser
+        @@begins_to_ends = {}
```

Comment:
https://docs.rubocop.org/rubocop/cops_style.html#styleclassvars

This rule doesn't seem to be enabled here but I generally agree that we should prefer singleton class instance variables over class variables. 
```suggestion
        @begins_to_ends = {}
```